pkg/tracing/tracer_otel.go

```diff
@@ -73,12 +77,19 @@ func (t *otelTracer) spanFromTrace(ctx context.Context, trc *Trace, name string)
 		tFlags = trace.FlagsSampled
 	}
 
+	// The Trace ID is expected to be compliant with the W3C trace-context specification. If it is not
+	// an empty traceID will be used for the new span.
 	tID, err := trace.TraceIDFromHex(trc.GetTraceId())
 	if err != nil {
+		t.logger.Warn("could not parse trace id %w", err)
```

Comment:
`%w` doesn't work when logging warnings, you want to use `%s` in this case (using `%w` results in something like `%!w(*errors.errorString=&{your error message}) ` because the arguments are passed to `fmt.Sprintf` and not `fmt.Errorf` for non-error levels)